ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/RaceConditionTests.scala

```diff
@@ -107,7 +107,7 @@ private[testtool] object RaceConditionTests {
       tx.hasEventsNumber(1) &&
         tx.containsEvent(isExerciseEvent(RaceTests.ContractWithKey.ChoiceArchive))
 
-    def isNonConsumingExercise(tx: TransactionTree): Boolean =
+    def isExercise(tx: TransactionTree): Boolean =
```

Comment:
I’m a bit confuse. Afaict we filter by the choice name and the choice is non-consuming https://github.com/digital-asset/daml/blob/800f3741024e176eb9bdc1a1d08c18c01e5bee67/ledger/test-common/src/main/daml/semantic/RaceTests.daml#L26 Why are you suggesting that it’s not non-consuming?

Reply:
Oh you're right, but then we have a problem in `RWArchiveVsRollbackConsumingChoice` I believe

Reply:
good catch, will look into it today if you don’t beat me to it.

Reply:
actually I think the problem is much bigger. All the filters in there seem crap :cry:

Reply:
Kool